gwpy/table/table.py

```diff
@@ -779,14 +779,23 @@ def cluster(self, index, rank, window):
         if window <= 0.0:
             raise ValueError('Window must be a positive value')
 
+        # If no rows, no need to cluster
+        if len(self) == 0:
+            return self
```

Comment:
I like this short-circuit, however the documented behvaiour is to return a _new_ table, so we might need to make a copy:

```suggestion
            return self.copy()
```